style/hudl_sql_style_guide.md

```diff
@@ -5,33 +5,101 @@ description: A guide to writing clean, clear, and consistent SQL.
+
+When it has been determined that a new data model is required to be built, the person building the new model will be required to hold a data model design review. This is a process in which the builder of the data model will write-up their design plans and hold a meeting for these plans to be peer reviewed. 
+
+The purpose of this is not to block or slow work. Rather, this is a means to ensure data quality, prevent painful re-work and identify dependencies that may have been overlooked. Each data model design will require the following:
+
+* __State the purpose__: 
+    Every new data model that is introduced must have a clear purpose linked to business value. All models added should be done so with a high amount of rigor to       ensure data quality.
+* __Start With the End in Mind__: 
+    Test Driven Development https://www.guru99.com/test-driven-development.html is when the tests are written before the code. While we don’t require a full set of     all tests in the Data Model Design process, an overview of test cases that will be considered and implemented should be included as a part of the data model       design review.
+* __Identify the Grains__: 
+    Identify what the grains will be of the final table and of any tables from which you will be creating your final table.
+* Sample of table output with examples (can be hypothetical)
+* Column names and descriptions defined 
+
+### File Orgainzation
+* __Business-centric__ models and __source-centric__ models should be organizationally distinct. __Business-centric__ models define business logic. __Source-centric__ models select from dbt sources and their primary function is to clean and group like-data (ie: payment data may come from two different sources, but need to be joined).
+
+  __Business-centric__ models should be created within the marts directory. __Source-centric__ models should be created within the staging directory. 
+
+> Many pre-existing business-centric models do not exist within the marts directory.
+            However, all new ones should be created there.
+
+* Define raw data as dbt sources: No raw data sources should be referenced other than in sources. If a source and staging layer do not exist for a model you want to build, you are required to build the needed sources and staging layers.
+* __Staging__ 
+  - Used to clean sources
+  - Rename and Recast fields in this layer
+  - There should only be one staging file per one source. Only seed files should be joined at this layer if necessary.
+  - No business metrics/values should be created at this layer
+
+* __dbt seed__
+  - All seed files belong in the `data` directory.
+  - Seed files are best used for mappings (ie. country codes to country names)
+* __dbt snapshots__
+  - Use to track slowly changing dimensions
+  - As a general rule should be built on top of sources
+* __dbt macros__
+  - Best for trusted, reusable pieces of logic (ie.fiscal year creation).
```

Comment:
These are very dbt-centric. Wondering if we want these as part of the modeling best practices as they could change if we were to ever change tools?